### PR TITLE
Feature/scrum 80 extend signup form

### PR DESCRIPTION
**Pull request by holotsvan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@typescript-eslint/eslint-plugin": "^6.14.0",
     "@typescript-eslint/parser": "^6.17.0",
     "@vitejs/plugin-react": "^4.2.1",
-    "eslint": "^8.55.0",
+    "eslint": "^8.56.0",
```
Comment (Misha2048) [inline package.json]: shit, motherfucka

**Review by PavloDolia — APPROVED** [review]

File: src/components/AutoComplete.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+const AutoComplete: React.FC<IAutoComplete> = ({
+  options,
+  label = 'Input',
+  name,
+  required,
+  type,
+  value,
+  onChange,
+}) => {
+  const autoCompleteRef = useRef<any>()
```
Comment (Misha2048) [inline src/components/AutoComplete.tsx]: any type is inappropriate, do you undastand?

File: src/components/regestration/SignUp.tsx [hunk 3]
```diff
@@ -11,40 +12,55 @@ import { Title, DisplayTitle } from '../Titles'
...
 import Logo from '../Logo'
 import { api } from '../../api'
-import { ISignUpRequest } from '../../api/intefaces'
+import { ISignUpRequest, IUniversity } from '../../api/intefaces'
 import ToolTip from '../ToolTip'
+import AutoComplete from '@components/AutoComplete'
 
 function SignUp() {
-  //Referencing to SignUpFormState interface
+  const [universities, setUniversities] = useState<Array<IUniversity>>([])
```
Comment (Misha2048) [inline src/components/regestration/SignUp.tsx]: Why not IUniversity[]???

File: src/components/regestration/SignUp.tsx [hunk 4]
```diff
@@ -11,40 +12,55 @@ import { Title, DisplayTitle } from '../Titles'
...
     first_name: '',
     last_name: '',
     email: '',
     password: '',
+    university_id: '',
   })
 
+  useEffect(() => {
+    const universitiesList = async () => {
+      const unilist = await api.universities.getUniversities().then((data) => data)
```
Comment (Misha2048) [inline src/components/regestration/SignUp.tsx]: await?

**Review by Misha2048 — APPROVED** [review]
Smells like shit, but not much

**Review by PavloDolia — APPROVED** [review]
If it doesn't work, a bug report will be created later.

**Pull request merged**